Sources/AMQPClient/ChannelHandlers/AMQPChannelHandler.swift

```diff
@@ -11,421 +11,214 @@
-
-            switch msg.frame {
-            case .getOk(let getOk):
-                if let promise = self.responseQueue.popFirst() {
-                        promise.succeed(.channel(.message(.get(.init(
-                            message: .init(
-                                exchange: getOk.exchange,
-                                routingKey: getOk.routingKey,
-                                deliveryTag: getOk.deliveryTag,
-                                properties: properties,
-                                redelivered: getOk.redelivered,
-                                body: body),
-                            messageCount: getOk.messageCount)))))
-                }
-            case .deliver(let deliver):
-                self.notify(type: Delivery.self,
-                            named: deliver.consumerTag,
-                            .success(.init(
-                                exchange: deliver.exchange,
-                                routingKey: deliver.routingKey,
-                                deliveryTag: deliver.deliveryTag,
-                                properties: properties,
-                                redelivered: deliver.redelivered,
-                                body: body)))
-            case .return(let `return`):
-                self.notify(type: Return.self,
-                            .success(.init(
-                                replyCode: `return`.replyCode,
-                                replyText: `return`.replyText,
-                                exchange: `return`.exchange,
-                                routingKey: `return`.routingKey,
-                                properties: properties,
-                                body: body)))
-            default:
-                preconditionUnexpectedPayload(payload)
-            }
-
-            nextMessage = nil
-        default:
-            preconditionUnexpectedPayload(payload)
+    func handleCancellation(consumerTag: String) {
+        if let listener = listeners.withLockedValue({ $0.consumeListeners[consumerTag] }) {
+            listener(.failure(AMQPConnectionError.consumerCancelled))
         }
+        removeListener(type: Delivery.self, named: consumerTag)
     }
 
-    func close(error: Error? = nil) {
-        let shouldClose = self.state.withLockedValue { state in
+    func reportAsClosed(error: Error? = nil) {
+        let shouldClose = state.withLockedValue { state in
             if state == .open {
                 state = .shuttingDown
                 return true
             }
 
             return false
         }
-        
-        guard shouldClose else { return }
-
-        let queue = self.responseQueue
-        self.responseQueue.removeAll()
-
-        queue.forEach { $0.fail(error ?? AMQPConnectionError.channelClosed()) }
 
-        self.notify(type: Close.self, error == nil ? .success(()) : .failure(error!) )
+        guard shouldClose else { return }
 
-        self.consumeListeners = [:]
```

Comment:
This was based on https://github.com/swift-server-community/mqtt-nio/blob/acab3749ef1facf18aa25435dd30c246c374935b/Sources/MQTTNIO/MQTTClient.swift#L223.
Maybe we can add removeAll method on  AMQPChannels like them?

Reply:
since I move all event-handler dictionaries into a locked box, I exchanged it with this.
```
        listeners.withLockedValue {
            $0.closeListeners.values.forEach { $0(error == nil ? .success(()) : .failure(error!)) }
            $0 = .init()
        }
```
should do the same basically (ie: let go of the memory/handler references)